druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java

```diff
@@ -2156,6 +2156,308 @@ public RelNode apply(RelBuilder b) {
+            + "'context':{'skipEmptyBuckets':true}}";
+
+    sql(sql).queryContains(druidChecker(expectedQuery));
+  }
+
+  /**
+   * Test to ensure that an aggregate with a filter clause in the presence of another aggregate
+   * without a filter clause does not have it's filter factored out into the outer filter
+   * */
+  @Test public void testFilterClauseNotFactoredOut1() {
+    String sql = "select sum(\"store_sales\") filter (where \"store_state\" = 'CA'), "
+            + "sum(\"store_cost\") from \"foodmart\"";
+    String expectedQuery = "{'queryType':'timeseries','dataSource':'foodmart','descending':false,"
+            + "'granularity':'all','aggregations':[{'type':'filtered','filter':{'type':'selector',"
+            + "'dimension':'store_state','value':'CA'},'aggregator':{'type':'doubleSum','name':"
+            + "'EXPR$0','fieldName':'store_sales'}},{'type':'doubleSum','name':'EXPR$1','fieldName'"
+            + ":'store_cost'}],'intervals':['1900-01-09T00:00:00.000/2992-01-10T00:00:00.000'],"
+            + "'context':{'skipEmptyBuckets':true}}";
+
+    sql(sql).queryContains(druidChecker(expectedQuery));
+  }
+
+  /**
+   * Test to ensure that an aggregate with a filter clause in the presence of another aggregate
+   * without a filter clause, and an outer filter does not have it's
+   * filter factored out into the outer filter
+   * */
+  @Test public void testFilterClauseNotFactoredOut2() {
+    String sql = "select sum(\"store_sales\") filter (where \"store_state\" = 'CA'), "
+            + "sum(\"store_cost\") from \"foodmart\" where \"the_year\" >= 1997";
+    String expectedQuery = "{'queryType':'timeseries','dataSource':'foodmart','descending':false,"
+            + "'granularity':'all','filter':{'type':'bound','dimension':'the_year','lower':'1997',"
+            + "'lowerStrict':false,'ordering':'numeric'},'aggregations':[{'type':'filtered',"
+            + "'filter':{'type':'selector','dimension':'store_state','value':'CA'},'aggregator':{"
+            + "'type':'doubleSum','name':'EXPR$0','fieldName':'store_sales'}},{'type':'doubleSum',"
+            + "'name':'EXPR$1','fieldName':'store_cost'}],"
+            + "'intervals':['1900-01-09T00:00:00.000/2992-01-10T00:00:00.000'],"
+            + "'context':{'skipEmptyBuckets':true}}";
+
+    sql(sql).queryContains(druidChecker(expectedQuery));
+  }
+
+  /**
+   * Test to ensure that multiple aggregates with filter clauses have their filters extracted to
+   * the outer filter field for data pruning
+   * */
+  @Test public void testFilterClausesFactoredForPruning1() {
+    String sql = "select "
+            + "sum(\"store_sales\") filter (where \"store_state\" = 'CA'), "
+            + "sum(\"store_sales\") filter (where \"store_state\" = 'WA') "
+            + "from \"foodmart\"";
+    String expectedQuery = "{'queryType':'timeseries','dataSource':'foodmart','descending':false,"
```

Comment:
I've added code the execute the queries and check their results. Let me know if there's anything else you'd like me to add/remove.